src/Console/ConsoleOptionParser.php

```diff
@@ -677,18 +679,19 @@ public function subcommands()
      * to parse the $argv
      *
      * @param array $argv Array of args (argv) to parse.
+     * @param \Cake\Console\ConsoleIo|null $io A ConsoleIo instance or null. If null prompt options will error.
      * @return array [$params, $args]
      * @throws \Cake\Console\Exception\ConsoleException When an invalid parameter is encountered.
      */
-    public function parse(array $argv): array
+    public function parse(array $argv, ?ConsoleIo $io): array
```

Comment:
optional Argument is BC, but not passing it will cause the `prompt` to be ignored

Reply:
Yes. We could raise an error if a prompt is used with no `ConsoleIo` so the failure isn't silent.

Reply:
could we have deprecation warning if user not pass it ?
and making required Argument in cake 5?

Reply:
yes, I plan on making this parameter required in 5.x